### ISSUE TITLE
(FEATURE REQUEST) Ability to cancel update

### ISSUE THREAD

**Issue by MasterGamingYT** [RocketRobz/TWiLightMenu-Updater#54]
Is it possible that you can add a feature that cancels the pending update and deletes the 7z file?

**Comment by legoj15** [comment]
IMO I don't think this is too bad of an Idea. But it'd probably have to be implemented so that when the extraction/installation phase is running, it's not cancelable (could cause issues if cancled then).

**Comment by MasterGamingYT** [comment]
Ok, maybe the update can only be cancelled during the downloading phase

**Comment by FlameKat53** [comment]
Yeah, only cancelling during download should be implemented, during extracting would mess stuff up

**Comment by MasterGamingYT** [comment]
^

**Comment by MasterGamingYT** [comment]
Hope this request becomes a thing

**Comment by NightYoshi370** [comment]
Why should there be an option to cancel the update?

**Comment by MasterGamingYT** [comment]
It's for people who want to stay on an older version

**Comment by eku** [comment]
@RocketRobz If the download hangs in the middle of what I am watching, then it cannot be aborted. The udpater even blocks the HOME menu and you can only revive the 3DS with the power button.

For me this is more a *bug* than a feature request.

**Comment by SuperSaiyajinStackZ** [comment]
I think that's not really fixable, cause the 3DS is too weak iirc.

**Comment by RocketRobz** [comment]
It should be possible though. Just have a thread where it checks for a button press, and then when nothing is being downloaded or extracting, then the update will be canceled.
